src/containers/AddPaymentDetailsFinalizationHandler/AddPaymentDetailsFinalizationHandler.js

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  updatePaymentDetailsPopup,
+  PAYMENT_DETAILS_STEPS
+} from 'redux/popupSlice';
+import { setActiveTab, MYACCCOUNT_TABS } from 'redux/myaccountSlice';
+
+const AddPaymentDetailsFinalizationHandler = Component => {
```

Comment:
as for initial payment, we could rename it to withAddPaymentDetailsFinalizationHandler